df.history.xml

```diff
@@ -794,21 +803,20 @@
         <int32_t name='race' ref-target='creature_raw'/>
         <int16_t name='caste' ref-target='caste_raw' aux-value='$$.race'/>
 
-        <int32_t/>
-        <int32_t name='histfig_id' ref-target='historical_figure'
-                 comment='when masquerading as another historical figure'/>
-        <int32_t name='unk_4c'/>
+        <int32_t name='unk_1'/>
+        <int32_t name='histfig_nemesis_id' comment='reference depending on type (below): historical_figure for HidingCurse, TrueName and Identity, nemesis_record for FalseIdentity'/>
```

Comment:
Maybe this could be a union instead? It would be somewhat more overhead, but I believe the new tagged union support would allow you to associate multiple values of `type` with either `histfig_id` or `nemesis_id` - like [this](https://github.com/DFHack/df-structures/blob/master/df.units.xml#L1865-L1964), but with multiple enum attrs pointing to the same field (@BenLubar can correct me if this won't work). I think it would end up adding an intermediate compound, though, like `identity.ids.histfig_id` - not sure of a good name for `ids`.